lib/handover.js

```diff
@@ -48,13 +48,17 @@ class Plugin {
         case 'eth.':
           if (labels.length < 2) {
             return this.sendSOA();
-          } else {
-            data = await this.ethereum.resolveDnsFromEns(name, type);
-            if (!data || data.length === 0)
-              return this.sendSOA();
+          }
 
-            return this.sendData(data, type);
+          try {
+            data = await this.ethereum.resolveDnsFromEns(name, type);
+            if (data && data.length > 0)
+              return this.sendData(data, type);
+          } catch (e) {
+            this.logger.warning('Resolution failed for name: %s', name);
+            this.logger.debug(e.stack);
           }
+          return this.sendSOA();
```

Comment:
You could leave the `catch` empty here and log the warning before `return this.sendSOA()`. I think we can drop the stack trace too. Same with `._eth`, if we don't get an answer we can log "resolution failed" and maybe that's all we need to do.